Add federated services docs (#1874)
linkerd/website: linkerd.io/content/2-edge/features/multicluster.md

* Better multi-cluster authorization policies, as workload identity
  is preserved across cluster boundaries.

### Federated services

A federated service is a union of services with the same name and namespace
in multiple different clusters. Meshed clients that send traffic to a federated
service will have that traffic distributed across all replicas of services in
the federated service across clusters. Federated services use the *flat
networking* model and do not use a gateway intermediary.

> These modes can be combined, with each specific services by using the mode that
> is most appropriate. See the

Reviewer: ```suggestion
These modes can be combined, with each specific services selecting the mode that
is most appropriate for that service. See the
```

By default, Linkerd will mirror all exported services as Kubernetes `clusterIP`
services. This also extends to [headless services][headless-svc]; an exported
headless service will be mirrored as `clusterIP` and have an IP address

> assigned to it. In general, headless services *should not have an IP address*,
> they are used when a workloads needs a stable network identifier or to

Reviewer: ```suggestion
assigned to it. In general, headless services *should not have an IP address*;
they are used when a workload needs a stable network identifier or to
```

(I know the second line here isn't something you changed, I just noticed it.)
